File: 프로그래머스/2/49993. 스킬트리/스킬트리.py
def solution(skill, skill_trees):
    answer = 0
    for tree in skill_trees:
        filtered_skill = [s for s in tree if s in skill]
        if "".join(filtered_skill) == skill[:len(filtered_skill)]:
        # 앞을 배워야 뒤를 배울 수 있기 때문에 반드시 맨 처음부터 포함되어야 함
            answer += 1
            
    return answer

File: 프로그래머스/2/49993. 스킬트리/스킬트리.java
class Solution {
    public int solution(String skill, String[] skill_trees) {
        int answer = 0;
        for (String tree : skill_trees) {
            String filtered_skills = "";
            for (int i=0; i<tree.length(); i++) {
                if (skill.contains(tree.charAt(i)+""))
                    filtered_skills += tree.charAt(i);
            }
            if (filtered_skills.equals(skill.substring(0, filtered_skills.length())))
                answer++;
        }
        return answer;
    }
}